### ISSUE TITLE
Rename `singleGroups` option

### ISSUE THREAD

**Issue by jugglinmike** [jshint/jshint#2043]
The `singleGroups` option is the only JSHint option to use camel case. For consistency's sake, it ought to be named `singlegroups` instead. This is a breaking change, so it should be deferred until the 3.0 release of JSHint.

**Comment by jugglinmike** [comment]
@caitp @rwaldron Since we expanded the scope of the feature in gh-2109, I think a new name (instead of a change in character case) is in order. New proposal: `strictgroups`. What do you think?

**Comment by rwaldron** [comment]
Is that the name we came up with last week? Along with unaryGroup?

**Comment by jugglinmike** [comment]
It is. The new option we discussed last week was `unaryGroups`, but that should *also* be downcased to `unarygroups` for consistency. Issue here: gh-2102

But come to think of it, I think that new option should be merged into this one. As far as I can see, the proposed `unarygroups` is just a slightly less strict version of the current `singleGroups` behavior. New proposal:

- `strictgroups: true`: warn on any unnecessary usage of the grouping operator (i.e. the current behavior of `singleGroups: true`)
- `strictgroups: unary`: the behavior proposed for `unarygroups` in gh-2102
- `strictgroups: false`: all grouping-operator-related warnings disabled

Howsabout that?

**Comment by rwaldron** [comment]
Seems good

**Comment by andyburke** [comment]
Would setting strictgroups: unary in this new system allow for constructions like:

```javascript
var blah = foo || ( bar && baz );
```

Currently, this produces a warning with singleGroups set to true. The grouping in this case may not be strictly necessary, but can be helpful from a readability standpoint. And really, I just want to disallow things like delete( foo[ 'bar' ] ) and typeof( blah ).

If the strictgroups: unary would still emit a warning with the above, then I will open a separate issue to address that.

**Comment by jugglinmike** [comment]
That's correct. "strictgroups: unary" would allow `var blah = foo || ( bar && baz );` but issue warnings for `delete( foo[ 'bar' ] )` and `typeof( blah )`.

**Comment by andyburke** [comment]
:trophy:

**Comment by jugglinmike** [comment]
:cactus:

**Comment by cr-nz** [comment]
What version of JSHint do I need to use to be able to use `strictgroups: unary`?

Currently I have `"singleGroups": true,` enabled but it's not very useful, i.e. it is throwing many warnings across the code base on things like:

`if ((page === 'contact') || (page === 'terms')) {`

I prefer to keep the () around each condition for readability. I would prefer if the JSHint was only throwing warnings on things like `typeof(varName)`.
